ton_client/client/src/api/mod.rs

```diff
@@ -0,0 +1,309 @@
+pub(crate) mod interop;
+mod net;
+pub(crate) mod processing;
+
+use crate::abi::{
+    attach_signature, decode_message, encode_message, Abi, AbiHandle, CallSet, DeploySet,
+    FunctionHeader,
+};
+use crate::api::api_reference::get_api_reference;
+use crate::api::dispatch::{ApiDispatcher, ModuleReg, Registrar};
+use crate::boc::{
+    get_blockchain_config, parse_account, parse_block, parse_message, parse_transaction,
+};
+use crate::crypto::{
+    convert_public_key_to_ton_safe_format, factorize, generate_random_bytes,
+    generate_random_sign_keys, hdkey_derive_from_xprv, hdkey_derive_from_xprv_path,
+    hdkey_public_from_xprv, hdkey_secret_from_xprv, hdkey_xprv_from_mnemonic,
+    mnemonic_derive_sign_keys, mnemonic_from_entropy, mnemonic_from_random, mnemonic_verify,
+    mnemonic_words, modular_power, nacl_box, nacl_box_keypair, nacl_box_keypair_from_secret_key,
+    nacl_box_open, nacl_secret_box, nacl_secret_box_open, nacl_sign, nacl_sign_detached,
+    nacl_sign_keypair_from_secret_key, nacl_sign_open, sha256, sha512, sign, ton_crc16,
+    verify_signature,
+};
+use crate::net::{query_collection, unsubscribe, wait_for_collection};
+use crate::processing::{DecodedOutput, MessageSource, ProcessingEvent, ResultOfProcessMessage};
+use crate::tvm::{execute_get, execute_message};
+use crate::utils::{convert_address, AddressStringFormat};
+
+lazy_static! {
+    static ref DISPATCHER: ApiDispatcher = create_dispatcher();
+}
+
+pub(crate) fn get_dispatcher() -> &'static ApiDispatcher {
+    &DISPATCHER
+}
+
+/// BOC manipulation module.
+#[derive(ApiModule)]
+#[api_module(name = "client")]
+pub(crate) struct ClientModule;
+
+impl ModuleReg for ClientModule {
+    fn reg(reg: &mut Registrar) {
+        reg.f_no_args(
+            get_api_reference,
+            crate::api::api_reference::get_api_reference_api,
+        );
+        reg.f_no_args(crate::client::version, crate::client::version_api);
+    }
+}
+
+/// Crypto functions.
+#[derive(ApiModule)]
+#[api_module(name = "crypto")]
+pub(crate) struct CryptoModule;
+
+impl ModuleReg for CryptoModule {
+    fn reg(reg: &mut Registrar) {
+        // Math
+
+        reg.f(factorize, crate::crypto::math::factorize_api);
+        reg.f(modular_power, crate::crypto::math::modular_power_api);
+        reg.f(ton_crc16, crate::crypto::math::ton_crc16_api);
+        reg.f(
+            generate_random_bytes,
+            crate::crypto::math::generate_random_bytes_api,
+        );
+
+        // Keys
+
+        reg.f(
+            convert_public_key_to_ton_safe_format,
+            crate::crypto::keys::convert_public_key_to_ton_safe_format_api,
+        );
+
+        reg.f_no_args(
+            generate_random_sign_keys,
+            crate::crypto::keys::generate_random_sign_keys_api,
+        );
+        reg.f(sign, crate::crypto::keys::sign_api);
+        reg.f(verify_signature, crate::crypto::keys::verify_signature_api);
+
+        // Sha
+
+        reg.f(sha256, crate::crypto::hash::sha256_api);
```

Comment:
Why first parameter is used by short name but second - with fully qualified? I think Fully qualified names are better. They are more representative and `use` section will be shorter

Reply:
qualified